app/helpers/form_helper.rb

```diff
@@ -198,17 +198,14 @@ def file_field_f(f, attr, options = {})
   end
 
   def autocomplete_f(f, attr, options = {})
-    field(f, attr, options) do
-      path = options.delete(:path) || send("#{f.object.class.pluralize.underscore}_path") if options[:full_path].nil?
-      auto_complete_search(attr,
-                           f.object.send(attr).try(:squeeze, " "),
-                           options.merge(
-                             :placeholder => _("Filter") + ' ...',
-                             :path        => path,
-                             :name        => "#{f.object_name}[#{attr}]"
-                           )
-      ).html_safe
-    end
+    render('common/form_autocomplete', f: f,
+                                       name: options[:name] || "#{f.object_name}[#{attr}]",
+                                       controller: options[:controller] || auto_complete_controller_name,
+                                       url: options[:full_path] || "#{options[:path]}/auto_complete_search" || '',
+                                       isDisabled: options[:disabled] || false,
+                                       useKeyShortcuts: options[:use_key_shortcuts] || false,
+                                       initialQuery: options[:query] ||  options[:search_query] || f.object.search || '',
+                                       id: options[:id] || options[:control_group_id])
```

Comment:
I agree with what you are saying here, 
though the original `autocomplete_f` was used this way also in plugins 
and this is why we have to support all of the old options.

Lets look at all of 10 options:
`name`, `controller`, `full_path`, `path`, `disabled`, `use_key_shortcuts`, `query`, `search_query`, `id`, `control_group_id`

eventually, this options are transformed into 7 options due to duplication.
Currently I don't agree that we can delete some of them,
What we can do is start to inform of deprecation, and let plugins fix their options too,
or we can do it for them, but I don't think it is in the scope of this PR.

`size` isn't used here anymore so I removed it.
can `control_group_id` changed and be used in another place? yes.. so I think we should keep it.

I can send the options hash into the `common/form_autocomplete` file on render, 
but than I will need to do the same work and it will get messy,
currently the `common/form_autocomplete` looks like this:
```
<div id="<%= id %>"></div>
  <%=mount_react_component('ComponentWrapper', "##{id}", {
    component: 'FormAutocomplete',
    componentProps: {
      url: url,
      isDisabled: isDisabled,
      controller: controller,
      useKeyShortcuts: useKeyShortcuts,
      initialQuery: initialQuery,
      name: name,
      id: id
    }
  }.to_json) %>
```

Reply:
thanks for the explanation !
refactored all options and it is still working as expected,
could you check also with plugins, my vagrant-stack doesn't want to work today.

as for the `options[:error]` i tried to implement it and pass an invalid argument, 
there was client error while trying to search that term,
but there was never a server-error,
could this be related to some plugin, do you know where it was used? thanks

Reply:
right, the error doesn't come from the options parameter, it comes from `f.object.errors[attr]`

Reply:
Added and tested :)